// react-sanfona/dist-modules/AccordionItemTitle/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _typeof = typeof Symbol === "function" && typeof Symbol.iterator === "symbol" ? function (obj) { return typeof obj; } : function (obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; };

exports.default = AccordionItemTitle;

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function AccordionItemTitle(_ref) {
  var className = _ref.className,
      expanded = _ref.expanded,
      onClick = _ref.onClick,
      Root = _ref.rootTag,
      title = _ref.title,
      uuid = _ref.uuid;

  var style = {
    cursor: 'pointer',
    margin: 0
  };

  if ((typeof title === 'undefined' ? 'undefined' : _typeof(title)) === 'object') {
    return _react2.default.cloneElement(title, {
      onClick: onClick,
      id: 'react-safona-item-title-' + uuid,
      'aria-controls': 'react-sanfona-item-body-' + uuid
    });
  }

  return _react2.default.createElement(
    Root,
    {
      'aria-controls': 'react-sanfona-item-body-' + uuid,
      'aria-expanded': expanded,
      className: (0, _classnames2.default)('react-sanfona-item-title', className),
      id: 'react-safona-item-title-' + uuid,
      onClick: onClick,
      style: style
    },
    title
  );
}

AccordionItemTitle.defaultProps = {
  rootTag: 'h3'
};

AccordionItemTitle.propTypes = {
  className: _propTypes2.default.string,
  expanded: _propTypes2.default.bool,
  onClick: _propTypes2.default.func,
  rootTag: _propTypes2.default.string,
  title: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.node]),
  uuid: _propTypes2.default.string
};
